gui/default/syncthing/core/pathIsSubDirDirective.js

```diff
@@ -6,7 +6,11 @@ angular.module('syncthing.core')
                 ctrl.$validators.folderPathErrors = function (viewValue) {
                     // This function checks whether ydir is a subdirectory of xdir,
                     // e.g. it would return true if xdir = "/home/a", ydir = "/home/a/b".
+                    // Tildes in both xdir and ydir are expanded for comparison
+                    // so that e.g. xdir = "home/a/", ydir = "~/b" will return true.
                     function isSubDir(xdir, ydir) {
+                        xdir = xdir.replace("~", scope.system.tilde);
```

Comment:
I don't think we want `/home/foo/my~dir/` to get expanded though, so this should probably be limited to replacing tilde followed by slash (and maybe backslash) at the beginning of the string?

Reply:
You're right. I adapted the replace call. Thank you for the fast review btw! :)